### PR TITLE
Unzip and untar .conda packages

### LINKED ISSUE #296: Support .conda files
https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/packages.html#:~:text=conda%20file%20format%20was%20introduced,tar%20files.

> The .conda file format was introduced in conda 4.7 as a more compact, and thus faster, alternative to a tarball.

> The .conda file format consists of an outer, uncompressed ZIP-format container, with 2 inner compressed .tar files.

> For the .conda format's initial internal compression format support, we chose Zstandard (zstd). The actual compression format used does not matter, as long as the format is supported by libarchive. The compression format may change in the future as more advanced compression algorithms are developed and no change to the .conda format is necessary. Only an updated libarchive would be required to add a new compression format to .conda files.

### PR DESCRIPTION
**Pull request by mkitti**
Fix #296

### CODE REVIEW

**Comment by giordano** [comment]
A test would be great :slightly_smiling_face:

File: src/Prefix.jl [hunk 1]
```diff
@@ -335,6 +335,18 @@ function setup(source::SetupSource{ArchiveSource}, targetdir, verbose; tar_flags
...
+                @info "Extracting conda package $(basename(source.path))..."
+            end
+            # First unzip the .conda file
+            run(`unzip -q $(source.path)`)
+            # The .conda file contains an archive called pkg-*.tar.zst
+            # Replace initial hash with pkg, and change the file extension to obtain the name
+            pkg_name = replace(basename(source.path), r"^[a-z0-9]{64}-" => "pkg-", ".conda" => ".tar.zst")
+            pkg_source = SetupSource{ArchiveSource}(joinpath(targetdir, pkg_name), source.hash, source.target)
+            # Run setup again to untar the pkg binaries
+            setup(pkg_source, targetdir, verbose; tar_flags = tar_flags)
```
Comment (giordano) [inline src/Prefix.jl:349]: This is quite ingenious :grin:

**Comment by mkitti** [comment]
What happened with the unprivileged test?

**Comment by giordano** [comment]
Slow download, the job timed out

**Comment by mkitti** [comment]
Cool. Are we good then? I added tests. I streamlined the extraction.

Should we find a smaller package to download?

**Comment by giordano** [comment]
> Should we find a smaller package to download?

Yes, that'd be perfect

**Comment by mkitti** [comment]
How about https://anaconda.org/conda-forge/versionparsing.jl/1.3.0/download/noarch/versionparsing.jl-1.3.0-ha770c72_0.conda ? 22 kiB

**Comment by giordano** [comment]
Perfect!

**Comment by mkitti** [comment]
Done. Are we all set now?

**Pull request merged**